### PR TITLE
automated unit tests

### PR DESCRIPTION
**Pull request by uchchwhash**
success! [and evidence of failures hidden away!]

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          wget --no-verbose -O miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-py38_4.8.2-Linux-x86_64.sh
+
+          export CONDA_ROOT=${PWD}/conda
+          mkdir -p ${CONDA_ROOT}
+          chmod +x miniconda.sh && ./miniconda.sh -q -b -f -p ${CONDA_ROOT}
+          export PATH="${CONDA_ROOT}/bin:${PATH}"
+
+          conda install -y -c conda-forge click==7.1.2 gdal==3.1.4 numpy==1.21.0 blosc==1.21.0
+
+          pip install git+https://github.com/sixy6e/idl-functions.git
```
Comment (ga-ellis) [inline .github/workflows/test.yml:26]: Do the unit tests require idl-functions to be installed for them to run?

**Comment by ga-ellis** [comment]
My concern is with the dependency versions. Let's say, for instance, the modifications are made to enable GDAL 3.2+ to be installed and we bump up the GH Actions to reflect it. What is the impetus to update the dea-wagl-docker image to be inline with that?

My initial proposal would be to have a "bare bones" WAGL container that can be used for local dev and/or unit tests etc. Conceptually we could then FROM our-new-wagl-container:0.x.x in the dea-wagl-docker script and install only the extra dependencies for the other projects at that stage.

I don't know exactly how practical this would be, but running these installations across the top of a vanilla OS image as part of the GH Actions seems wasteful and prone to future issues.

File: .github/workflows/test.yml (hunk 1 above)
Comment (uchchwhash) [reply .github/workflows/test.yml:26]: Installing wagl does.

**Comment by uchchwhash** [comment]
Agreed with @ga-ellis, this is a very sensible approach. If you want to work on that bare-bones wagl container, go for it. I do not see this as an immediate concern for now, and as we want other things to happen sooner rather than later, I would prefer recording this in a ticket https://github.com/GeoscienceAustralia/wagl/issues/360 for now.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Test
```
Comment (truth-quark) [inline .github/workflows/test.yml:1]: What about renaming to "Run Unit Tests" or something as descriptive for the title?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+          export CONDA_ROOT=${PWD}/conda
+          mkdir -p ${CONDA_ROOT}
+          chmod +x miniconda.sh && ./miniconda.sh -q -b -f -p ${CONDA_ROOT}
+          export PATH="${CONDA_ROOT}/bin:${PATH}"
+
+          conda install -y -c conda-forge click==7.1.2 gdal==3.1.4 numpy==1.21.0 blosc==1.21.0
+
+          pip install git+https://github.com/sixy6e/idl-functions.git
+          pip install .
+          python setup.py test
```
Comment (truth-quark) [inline .github/workflows/test.yml:28]: The `test` option is deprecated:
https://setuptools.readthedocs.io/en/latest/userguide/commands.html#test-build-package-and-run-a-unittest-suite

The recommended solution there is to use `tox`, however `pytest` can be run from the command line. I can't recall if there's a preferred pattern though.

**Review by truth-quark — COMMENTED** [review]
This solution works, although there's the deprecated stuff I've mentioned in the comments.

I noticed a lot of residual warnings from the `wagl` code & `Fortran` compilation step in the action logs. How polished does this action need to be for current work?

File: .github/workflows/test.yml (hunk 2 above)
Comment (uchchwhash) [reply .github/workflows/test.yml:1]: 👍🏽 Go for it.

File: .github/workflows/test.yml (hunk 3 above)
Comment (uchchwhash) [reply .github/workflows/test.yml:28]: If `pytest` works, please use `pytest`.

**Comment by uchchwhash** [comment]
Yes, `python setup.py test` is deprecated, but that is what Josh Sixsmith knew to work, and it works!
Totally agreed that `pytest` is better. So I'll hand it over to you @truth-quark.

**Comment by uchchwhash** [comment]
I think we like @ga-ellis 's solution more. Closing this one.

**Pull request closed without merging**